doc/api/typescript.md

```diff
@@ -0,0 +1,153 @@
+# Modules: TypeScript
+
+## Enabling
+
+There are two ways to enable runtime TypeScript support in Node.js:
+
+1. For [full support][] of all of TypeScript's syntax and features, including
+   using any version of TypeScript, use a third-party package.
+
+2. For lightweight support, you can use the built-in support for
+   [type stripping][].
+
+## Full TypeScript support
+
+To use TypeScript with full support for all TypeScript features, including
+`tsconfig.json`, you can use a third-party package. These instructions use
+[`tsx`][] as an example but there are many other similar libraries available.
+
+1. Install the package as a development dependency using whatever package
+   manager you're using for your project. For example, with `npm`:
+
+   ```bash
+   npm install --save-dev tsx
+   ```
+
+2. Then you can run your TypeScript code via:
+
+   ```bash
+   npx tsx your-file.ts
+   ```
+
+   Or alternatively, you can run with `node` via:
+
+   ```bash
+   node --import=tsx your-file.ts
+   ```
+
+## Type stripping
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+The flag [`--experimental-strip-types`][] enables Node.js to run TypeScript
```

Comment:
I am not sure where it would go in this doc, but it would be good (somewhere) to inducate which TS language version the bundled transpiler supports, since users cannot upgrade this out of band and the `amaro` / `swc` versioning does not indicate that.

Reply:
this should definitely be specificied both in amaro and core, we will followup with doc/guide and better errors